ht.el

```diff
@@ -162,6 +162,14 @@ For every key-value pair in TABLE, evaluate FORM with the
 variables key and value bound."
   `(ht-each (lambda (key value) ,form) ,table))
 
+(defun ht-select-keys (table keys)
+  "Return a new TABLE according to specific KEYS."
+  (let (result)
+    (setq result (make-hash-table))
```

Comment:
In elisp, hash-tables have an equality predicate associated with them. We should probably do the same here.

(Saying that, I've just realised that this is a bug in the other functions too!)